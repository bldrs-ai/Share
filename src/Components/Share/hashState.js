/** The prefix to use for the Share state token */
export const HASH_PREFIX_SHARE = 'share'


/** @return {boolean} */
export function isVisibleInitially() {
  return false
}
