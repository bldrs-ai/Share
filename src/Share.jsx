import React, {createContext, useEffect, useState} from 'react'
import {useNavigate, useParams} from 'react-router-dom'
import {ThemeProvider} from '@mui/material/styles'
import CssBaseline from '@mui/material/CssBaseline'
import CadView from './Containers/CadView'
import useTheme from './Theme'
import debug from './utils/debug'
import './index.css'
// TODO: This isn't used.
// If icons-material isn't imported somewhere, mui dies
/* eslint-disable */
import AccountCircle from '@mui/icons-material/AccountCircle'
/* eslint-enable */


/**
 * Handles path demuxing to pass to CadView.
 * @param {string} installPrefix e.g. '' on bldrs.ai or /Share on GitHub pages.
 * @param {string} appPrefix e.g. /share is the prefix for this component.
 * @param {string} pathPrefix e.g. v/p for CadView, currently the only child.
 * @return {Object} The Share react component.
 */
export default function Share({installPrefix, appPrefix, pathPrefix}) {
  const navigate = useNavigate()
  const urlParams = useParams()
  const [modelPath, setModelPath] = useState(null)


  /**
   * On a change to urlParams, setting a new model path will clear the
   * scene and load the new model IFC.  If there's not a valid IFC,
   * the helper will redirect to the index file.
   *
   * Otherwise, the param change is a sub-path, e.g. the IFC element
   * path, so no other useEffect is triggered.
   */
  useEffect(() => {
    onChangeUrlParams()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [urlParams])


  /** A demux to help forward to the index file, load a new model or do nothing. */
  function onChangeUrlParams() {
    const mp = getModelPath(installPrefix, pathPrefix, urlParams)
    if (mp === null) {
      navToDefault(navigate, appPrefix)
      return
    }
    if (modelPath === null ||
        modelPath.filepath && modelPath.filepath != mp.filepath ||
        modelPath.gitpath && modelPath.gitpath != mp.gitpath) {
      setModelPath(mp)
      debug().log('Share#onChangeUrlParams: new model path: ', mp)
    }
  }


  const {theme, colorMode} = useTheme()

  return (
    modelPath &&
      <CssBaseline>
        <ColorModeContext.Provider value={colorMode}>
          <ThemeProvider theme={theme}>
            <CadView
              installPrefix={installPrefix}
              appPrefix={appPrefix}
              pathPrefix={pathPrefix}
              modelPath={modelPath}
            />
          </ThemeProvider>
        </ColorModeContext.Provider>
      </CssBaseline>)
}


/**
 * Navigate to index.ifc with nice camera setting.
 * @param {Object} navigate
 * @param {string} appPrefix
 */
export function navToDefault(navigate, appPrefix) {
  // TODO: probe for index.ifc
  if (window.innerWidth <= 900) {
    navigate(appPrefix + '/v/p/index.ifc#c:-144.36,14.11,147.82,-40.42,17.84,-2.28')
  } else {
    navigate(appPrefix + '/v/p/index.ifc#c:-111.37,14.94,90.63,-43.48,15.73,-4.34')
  }
}


/**
 * Returns a reference to an IFC model file.  For use by IfcViewerAPI.load.
 *
 * Format is either a reference within this project's serving directory:
 *   {filepath: '/file.ifc'}
 *
 * or a global GitHub path:
 *   {gitpath: 'http://host/share/v/gh/bldrs-ai/Share/main/index.ifc'}
 *
 * @param {string} installPrefix e.g. /share
 * @param {string} pathPrefix e.g. /share/v/p
 * @param {Object} urlParams e.g. .../:org/:repo/:branch/*
 * @return {Object}
 */
function getModelPath(installPrefix, pathPrefix, urlParams) {
  // TODO: combine modelPath methods into class.
  let m = null
  let filepath = urlParams['*']
  if (filepath == '') {
    return null
  }
  const parts = filepath.split('.ifc')
  filepath = '/' + parts[0] + '.ifc' // TODO(pablo)
  if (pathPrefix.endsWith('new') || pathPrefix.endsWith('/p')) {
    // * param is defined in ../Share.jsx, e.g.:
    //   /v/p/*.  It should be only the filename.
    // Filepath is a reference rooted in the serving directory.
    // e.g. /haus.ifc or /ifc-files/haus.ifc
    m = {
      filepath: filepath,
      eltPath: parts[1],
    }
    debug().log('Share#getModelPath: is a project file: ', m, window.location.hash)
  } else if (pathPrefix.endsWith('/gh')) {
    m = {
      org: urlParams['org'],
      repo: urlParams['repo'],
      branch: urlParams['branch'],
      filepath: filepath,
      eltPath: parts[1],
    }
    m.getRepoPath = () => `/${m.org}/${m.repo}/${m.branch}${m.filepath}`
    m.gitpath = `https://raw.githubusercontent.com${m.getRepoPath()}`
    debug().log('Share#getModelPath: is a remote GitHub file: ', m)
  } else {
    throw new Error('Empty view type from pathPrefix')
  }
  return m
}


export const ColorModeContext = createContext({toggleColorMode: () => {}})
