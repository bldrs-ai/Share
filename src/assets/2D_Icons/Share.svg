<svg width="46" height="46" viewBox="0 0 46 46" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M28.14 11.0086C28.3455 11.0375 28.5389 11.1249 28.697 11.2608L39.6107 20.428C39.8574 20.6369 40 20.9452 40 21.27C40 21.5951 39.8574 21.9033 39.6107 22.1123L28.697 31.2795C28.4841 31.456 28.2126 31.5447 27.9373 31.5278C27.6622 31.511 27.4033 31.3897 27.2132 31.1884C27.0229 30.9871 26.9153 30.7209 26.912 30.4428V26.7874C24.4513 26.811 15.9879 27.2829 9.80215 32.7346C9.56031 32.9441 9.23956 33.0362 8.92437 32.9871C8.60945 32.9377 8.33133 32.7523 8.16359 32.4788C7.99561 32.2056 7.95475 31.8721 8.05143 31.566C11.1077 22.0746 20.2937 17.8734 26.9114 16.0504V12.1086C26.9091 11.7914 27.0426 11.4886 27.2778 11.2778C27.5131 11.067 27.8276 10.9692 28.14 11.0086ZM29.0949 14.4463V16.8757C29.0949 17.1195 29.0147 17.3566 28.8665 17.5494C28.7183 17.7425 28.5107 17.8802 28.2763 17.9413C22.9288 19.3059 15.6806 22.3609 11.8602 28.5753C19.1691 24.2469 27.3668 24.5434 28.0606 24.5761C28.34 24.591 28.6034 24.7132 28.7957 24.9179C28.9883 25.1227 29.0954 25.3941 29.0951 25.6762V28.0942L37.2233 21.2764L29.0949 14.4463Z"/>
</svg>
