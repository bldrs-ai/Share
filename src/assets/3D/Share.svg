<svg width="46" height="46" viewBox="0 0 46 46" fill="none" xmlns="http://www.w3.org/2000/svg">
<circle cx="23" cy="23" r="23" fill="white"/>
<path d="M25.7841 16.125L31.2728 16.125L30.4887 16.9091L25 16.9091L25 11.4204L25.7841 10.6364L25.7841 16.125Z" fill="#C4C4C4"/>
<path d="M11.7841 25.4886L17.2728 25.4886L16.4887 26.2727L11 26.2727L11 20.7841L11.7841 20L11.7841 25.4886Z" fill="#C4C4C4"/>
<path d="M25.7841 33.8523L31.2728 33.8523L30.4887 34.6364L25 34.6364L25 29.1477L25.7841 28.3636L25.7841 33.8523Z" fill="#C4C4C4"/>
<rect x="31.2728" y="10.6364" width="5.57577" height="5.57577" transform="rotate(90 31.2728 10.6364)" fill="#848484"/>
<rect x="17.2728" y="20" width="5.57577" height="5.57577" transform="rotate(90 17.2728 20)" fill="#848484"/>
<rect x="31.2728" y="28.3636" width="5.57577" height="5.57577" transform="rotate(90 31.2728 28.3636)" fill="#848484"/>
<path d="M14 23L30 12" stroke="#848484" stroke-width="2"/>
<path d="M15 24L29 31" stroke="#848484" stroke-width="2"/>
</svg>
