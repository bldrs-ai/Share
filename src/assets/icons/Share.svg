<svg width="49" height="49" viewBox="0 0 49 49" xmlns="http://www.w3.org/2000/svg">
<path d="M36.7502 28.5831C35.544 28.5911 34.3545 28.8662 33.2672 29.3887C32.1799 29.9111 31.2219 30.668 30.4619 31.6048L20.0494 26.8069C20.539 25.3078 20.539 23.6918 20.0494 22.1927L30.4619 17.3948C31.6902 18.877 33.4026 19.8776 35.2969 20.2201C37.1912 20.5626 39.1455 20.225 40.8151 19.2668C42.4847 18.3086 43.7621 16.7914 44.4219 14.9831C45.0817 13.1747 45.0816 11.1913 44.4214 9.3831C43.7612 7.57484 42.4835 6.05791 40.8138 5.10001C39.1441 4.14211 37.1896 3.80483 35.2954 4.14768C33.4012 4.49054 31.689 5.4915 30.461 6.9739C29.2329 8.45631 28.568 10.3249 28.5836 12.2498C28.5897 12.7362 28.6375 13.2211 28.7265 13.6994L17.9465 18.6606C16.7972 17.5368 15.3418 16.7772 13.7627 16.4769C12.1836 16.1766 10.551 16.349 9.06943 16.9724C7.58786 17.5959 6.32321 18.6427 5.43394 19.9817C4.54466 21.3207 4.07031 22.8924 4.07031 24.4998C4.07031 26.1072 4.54466 27.6789 5.43394 29.0179C6.32321 30.3569 7.58786 31.4037 9.06943 32.0271C10.551 32.6506 12.1836 32.823 13.7627 32.5227C15.3418 32.2224 16.7972 31.4627 17.9465 30.3389L28.7265 35.3002C28.6375 35.7784 28.5897 36.2634 28.5836 36.7498C28.5836 38.365 29.0625 39.9439 29.9599 41.2869C30.8573 42.6299 32.1327 43.6767 33.625 44.2948C35.1173 44.9129 36.7593 45.0746 38.3435 44.7595C39.9277 44.4444 41.3828 43.6666 42.5249 42.5245C43.6671 41.3824 44.4449 39.9272 44.76 38.343C45.0751 36.7588 44.9134 35.1168 44.2953 33.6245C43.6771 32.1323 42.6304 30.8568 41.2874 29.9594C39.9444 29.0621 38.3655 28.5831 36.7502 28.5831ZM36.7502 8.16645C37.5579 8.16645 38.3473 8.40594 39.0188 8.85462C39.6903 9.3033 40.2137 9.94103 40.5228 10.6872C40.8318 11.4333 40.9127 12.2543 40.7551 13.0464C40.5976 13.8385 40.2087 14.5661 39.6376 15.1371C39.0665 15.7082 38.339 16.0971 37.5469 16.2547C36.7548 16.4122 35.9338 16.3314 35.1876 16.0223C34.4415 15.7132 33.8038 15.1899 33.3551 14.5184C32.9064 13.8469 32.6669 13.0574 32.6669 12.2498C32.6669 11.1668 33.0971 10.1282 33.8629 9.36243C34.6287 8.59666 35.6673 8.16645 36.7502 8.16645V8.16645ZM12.2502 28.5831C11.4426 28.5831 10.6532 28.3436 9.98167 27.895C9.31017 27.4463 8.78679 26.8085 8.47774 26.0624C8.16868 25.3163 8.08782 24.4953 8.24537 23.7032C8.40293 22.9111 8.79183 22.1835 9.36289 21.6124C9.93396 21.0414 10.6615 20.6525 11.4536 20.4949C12.2457 20.3374 13.0667 20.4182 13.8129 20.7273C14.559 21.0363 15.1967 21.5597 15.6454 22.2312C16.0941 22.9027 16.3336 23.6922 16.3336 24.4998C16.3336 25.5828 15.9034 26.6214 15.1376 27.3871C14.3718 28.1529 13.3332 28.5831 12.2502 28.5831ZM36.7502 40.8331C35.9426 40.8331 35.1532 40.5936 34.4817 40.145C33.8102 39.6963 33.2868 39.0585 32.9777 38.3124C32.6687 37.5663 32.5878 36.7453 32.7454 35.9532C32.9029 35.1611 33.2918 34.4335 33.8629 33.8624C34.434 33.2914 35.1615 32.9025 35.9536 32.7449C36.7457 32.5874 37.5667 32.6682 38.3129 32.9773C39.059 33.2863 39.6967 33.8097 40.1454 34.4812C40.5941 35.1527 40.8336 35.9422 40.8336 36.7498C40.8336 37.8328 40.4034 38.8714 39.6376 39.6371C38.8718 40.4029 37.8332 40.8331 36.7502 40.8331Z"/>
</svg>
