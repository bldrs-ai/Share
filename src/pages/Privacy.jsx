import React, {ReactElement} from 'react'
import Typography from '@mui/material/Typography'
import Link from '@mui/material/Link'
import Box from '@mui/material/Box'
import TitledLayout from '../layouts/TitledLayout'


/** @return {ReactElement} */
export default function Privacy() {
  return (
    <TitledLayout title='Privacy Policy'>
      <Box sx={{display: 'grid'}}>
        <Typography variant='body1'><strong>Effective Date:</strong> January 1, 2022</Typography>
        <Typography variant='body1'><strong>Last Updated:</strong> October 23, 2025</Typography>

        <Typography variant='body1'>
          Bldrs Share (“we,” “us,” “our”) provides open-source, browser-first tools for viewing and
          collaborating on 3D model files (e.g., IFC, STEP) using our Conway Engine. We design our software
          to minimize data collection and maximize user control.
        </Typography>

        <Typography variant='h1'>1. Information We Collect</Typography>
        <Typography variant='body1'>
          Bldrs Share is built to operate <strong>without centralized data storage</strong>. We do not host
          user content or model data by default. Depending on your configuration:
        </Typography>
        <Typography variant='body1'>
          <strong>Authentication data.</strong> When you sign in via Auth0/Google or another provider,
          we receive basic profile information (name, email, avatar) to create or identify your account.
        </Typography>
        <Typography variant='body1'>
          <strong>Model data.</strong> Models you open are processed locally in your browser using WebAssembly.
          If you connect third-party storage (e.g., Google Drive), we only request access to the specific files
          you select. We do not scan or store other files from your account.
        </Typography>
        <Typography variant='body1'>
          <strong>Telemetry (optional).</strong> If enabled in a hosted deployment, we may collect anonymized
          usage metrics (e.g., performance, feature use) to improve reliability and UX.
        </Typography>

        <Typography variant='h1'>2. Open-Source and Self-Hosted Use</Typography>
        <Typography variant='body1'>
          Most of our codebase (including Bldrs Share and Conway) is available under the
          <strong> GNU Affero General Public License (AGPL)</strong> on GitHub. If you deploy or modify our
          software, you are responsible for complying with applicable privacy laws and for communicating your
          own deployment’s data-handling practices to your users.
        </Typography>

        <Typography variant='h1'>3. How We Use Information</Typography>
        <Typography variant='body1'>
          We use limited account and usage data to authenticate users; enable integrations (e.g., importing from
          Google Drive); improve performance, security, and usability; and communicate important service updates.
        </Typography>

        <Typography variant='h1'>4. Data Storage and Security</Typography>
        <Typography variant='body1'>
          <strong>No persistent model storage by default.</strong> Model data is processed client-side and is not
          uploaded to our servers. Communication between your browser and our services uses HTTPS. Future, optional
          hosted collaboration features (e.g., Firestore for comments/sharing) may store limited user-generated
          data; if launched, this policy will be updated accordingly.
        </Typography>

        <Typography variant='h1'>5. Google User Data</Typography>
        <Typography variant='body1'>
          Our use of Google user data complies with the{' '}
          <Link href='https://developers.google.com/terms/api-services-user-data-policy' target='_blank' rel='noopener'>
            Google API Services User Data Policy
          </Link>
          , including Limited Use requirements. Access to Google Drive is used solely to open or save the files that
          you explicitly select.
        </Typography>

        <Typography variant='h1'>6. Your Choices</Typography>
        <Typography variant='body1'>
          You can disconnect Google access in your Google Account settings. You may request deletion of any stored
          account metadata (if applicable) by emailing <Link href='mailto:support@bldrs.ai'>support@bldrs.ai</Link>.
          You can also use open-source/self-hosted deployments without accounts, subject to your configuration.
        </Typography>

        <Typography variant='h1'>7. Children’s Privacy</Typography>
        <Typography variant='body1'>
          Our services are not directed to children under 13, and we do not knowingly collect personal information
          from them.
        </Typography>

        <Typography variant='h1'>8. Updates</Typography>
        <Typography variant='body1'>
          We may update this Privacy Policy as our services evolve. The “Last Updated” date will reflect the most
          recent changes.
        </Typography>

        <Typography variant='h1'>9. Contact</Typography>
        <Typography variant='body1'>
          Questions or privacy requests? Email{' '}
          <Link href='mailto:support@bldrs.ai'>support@bldrs.ai</Link> or visit{' '}
          <Link href='https://bldrs.ai' target='_blank' rel='noopener'>https://bldrs.ai</Link>.
        </Typography>
      </Box>
    </TitledLayout>
  )
}
