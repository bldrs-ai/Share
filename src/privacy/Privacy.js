import {
  getCookie as getCookiePrivate,
  getCookieBoolean as getCookieBooleanPrivate,
  setCookie as setCookiePrivate} from './functional'
import {recordEvent, isAnalyticsAllowed, setIsAnalyticsAllowed} from './analytics'
import debug from '../utils/debug'
import {assertDefined} from '../utils/assert'


export {recordEvent, isAnalyticsAllowed, setIsAnalyticsAllowed}


/**
 * @param {string} component
 * @param {string} name
 * @param {string} defaultValue
 * @return {boolean} value of the setting
 */
export function getCookie({component, name, defaultValue}) {
  return getCookiePrivate(name, defaultValue)
}


/**
 * @param {string} component
 * @param {string} name
 * @param {boolean} defaultValue
 * @return {boolean} value of the setting
 */
export function getCookieBoolean({component, name, defaultValue}) {
  assertDefined(component, name, defaultValue)
  const value = getCookieBooleanPrivate(name, defaultValue)
  if (value == undefined) {
    return defaultValue
  }
  debug().log('Privacy#getCookieBoolean: ', component, name, value)
  return value
}


/**
 * @param {string} component
 * @param {string} name
 * @param {string} value
 */
export function setCookie({component, name, value}) {
  setCookiePrivate(name, value)
}


/**
 * @param {string} component
 * @param {string} name
 * @param {boolean} value
 */
export function setCookieBoolean({component, name, value}) {
  assertDefined(component, name, value)
  setCookiePrivate(name, value)
}


/**
 * @param {boolean} isUsageEnabled
 * @param {boolean} isSocialEnabled
 */
export function setUsageAndSocialEnabled(isUsageEnabled, isSocialEnabled) {
  assertDefined(isUsageEnabled, isSocialEnabled)
  debug().log('Privacy#setUsageAndSocialEnabled: ', isUsageEnabled, isSocialEnabled)
  setCookieBoolean({component: 'cookies', name: 'usage', value: isUsageEnabled})
  setCookieBoolean({component: 'cookies', name: 'social', value: isSocialEnabled})
}


/**
 * @return {boolean} for social privacy level
 */
export function isPrivacySocialEnabled() {
  return (
    getCookieBoolean({
      component: 'privacy',
      name: 'social',
      defaultValue: true})
  )
}


/**
 * @return {boolean} for usage privacy level
 */
export function isPrivacyUsageEnabled() {
  return (
    getCookieBoolean({
      component: 'privacy',
      name: 'usage',
      defaultValue: true})
  )
}
