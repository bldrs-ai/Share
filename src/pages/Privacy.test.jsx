import React from 'react'
import {render} from '@testing-library/react'
import {HelmetThemeCtx} from '../Share.fixture'
import Privacy from './Privacy'


describe('Privacy', () => {
  it('renders', () => {
    const {container, getByText} = render(<Privacy/>, {wrapper: HelmetThemeCtx})
    const title = getByText('Privacy Policy')
    expect(title).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
