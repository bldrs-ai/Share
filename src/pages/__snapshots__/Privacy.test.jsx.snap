// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Privacy renders 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-page-background MuiPaper-rounded css-1qmtcj9-MuiPaper-root"
  >
    <div
      class="MuiPaper-root MuiPaper-page MuiPaper-rounded css-1deatua-MuiPaper-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1jey5kd-MuiTypography-root-MuiLink-root"
        href="/"
      >
        <div
          class="MuiBox-root css-1dukxs6"
        >
          <svg
            data-filename="src/assets/LogoBWithDomain.svg"
          />
        </div>
      </a>
    </div>
    <div
      class="MuiPaper-root MuiPaper-page MuiPaper-rounded css-1xgkhtk-MuiPaper-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
      >
        Privacy Policy
      </h1>
      <div
        class="MuiBox-root css-lgj0h8"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          <strong>
            Effective Date:
          </strong>
           January 1, 2022
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          <strong>
            Last Updated:
          </strong>
           October 23, 2025
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          Bldrs Share (“we,” “us,” “our”) provides open-source, browser-first tools for viewing and collaborating on 3D model files (e.g., IFC, STEP) using our Conway Engine. We design our software to minimize data collection and maximize user control.
        </p>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
        >
          1. Information We Collect
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          Bldrs Share is built to operate 
          <strong>
            without centralized data storage
          </strong>
          . We do not host user content or model data by default. Depending on your configuration:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          <strong>
            Authentication data.
          </strong>
           When you sign in via Auth0/Google or another provider, we receive basic profile information (name, email, avatar) to create or identify your account.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          <strong>
            Model data.
          </strong>
           Models you open are processed locally in your browser using WebAssembly. If you connect third-party storage (e.g., Google Drive), we only request access to the specific files you select. We do not scan or store other files from your account.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          <strong>
            Telemetry (optional).
          </strong>
           If enabled in a hosted deployment, we may collect anonymized usage metrics (e.g., performance, feature use) to improve reliability and UX.
        </p>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
        >
          2. Open-Source and Self-Hosted Use
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          Most of our codebase (including Bldrs Share and Conway) is available under the
          <strong>
             GNU Affero General Public License (AGPL)
          </strong>
           on GitHub. If you deploy or modify our software, you are responsible for complying with applicable privacy laws and for communicating your own deployment’s data-handling practices to your users.
        </p>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
        >
          3. How We Use Information
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          We use limited account and usage data to authenticate users; enable integrations (e.g., importing from Google Drive); improve performance, security, and usability; and communicate important service updates.
        </p>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
        >
          4. Data Storage and Security
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          <strong>
            No persistent model storage by default.
          </strong>
           Model data is processed client-side and is not uploaded to our servers. Communication between your browser and our services uses HTTPS. Future, optional hosted collaboration features (e.g., Firestore for comments/sharing) may store limited user-generated data; if launched, this policy will be updated accordingly.
        </p>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
        >
          5. Google User Data
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          Our use of Google user data complies with the
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1jey5kd-MuiTypography-root-MuiLink-root"
            href="https://developers.google.com/terms/api-services-user-data-policy"
            rel="noopener"
            target="_blank"
          >
            Google API Services User Data Policy
          </a>
          , including Limited Use requirements. Access to Google Drive is used solely to open or save the files that you explicitly select.
        </p>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
        >
          6. Your Choices
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          You can disconnect Google access in your Google Account settings. You may request deletion of any stored account metadata (if applicable) by emailing 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1jey5kd-MuiTypography-root-MuiLink-root"
            href="mailto:support@bldrs.ai"
          >
            support@bldrs.ai
          </a>
          . You can also use open-source/self-hosted deployments without accounts, subject to your configuration.
        </p>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
        >
          7. Children’s Privacy
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          Our services are not directed to children under 13, and we do not knowingly collect personal information from them.
        </p>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
        >
          8. Updates
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          We may update this Privacy Policy as our services evolve. The “Last Updated” date will reflect the most recent changes.
        </p>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-wih9vs-MuiTypography-root"
        >
          9. Contact
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17ggkq7-MuiTypography-root"
        >
          Questions or privacy requests? Email
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1jey5kd-MuiTypography-root-MuiLink-root"
            href="mailto:support@bldrs.ai"
          >
            support@bldrs.ai
          </a>
           or visit
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1jey5kd-MuiTypography-root-MuiLink-root"
            href="https://bldrs.ai"
            rel="noopener"
            target="_blank"
          >
            https://bldrs.ai
          </a>
          .
        </p>
      </div>
    </div>
  </div>
</div>
`;
